api/src/main/java/org/apache/iceberg/UpdateSchema.java

```diff
@@ -659,4 +659,15 @@ default UpdateSchema setIdentifierFields(String... names) {
   default UpdateSchema caseSensitive(boolean caseSensitive) {
     throw new UnsupportedOperationException();
   }
+
+  /**
+   * Rollback table's schema to a specific {@link Schema} identified by id.
+   *
+   * @param schemaId schema id to roll back table to.
+   * @return this for method chaining
+   * @throws IllegalArgumentException If the table has no schema with the given id
+   */
+  default UpdateSchema rollbackTo(int schemaId) {
+    throw new UnsupportedOperationException();
+  }
```

Comment:
per this particular comment : https://github.com/apache/iceberg/issues/5591#issuecomment-1222524262

Seems like its intention not to rollback to prev schema, any reason why it ok to do it now ? 

It has a possible mitigation for undelete with required column by read defaults mentioned, any reason why this over that ?